pkg/tempopb/tempo.proto

```diff
@@ -52,11 +52,18 @@ message TraceByIDRequest {
   string blockStart = 2;
   string blockEnd = 3;
   string queryMode = 5;
+  bool allowPartialTrace = 6;
 }
 
 message TraceByIDResponse {
   Trace trace = 1;
   TraceByIDMetrics metrics = 2;
+  enum TraceType {
+    COMPLETE = 0;
+    PARTIAL = 1;
+  }
+  TraceType traceType = 3;
```

Comment:
i'd like to not constrain this to only "type" b/c we may want to communicate more in the future through it. how about something very generic like "Status" and the enum:

```
 enum Status {
    None = 0;
    PartialTrace = 1;
  }
```

Reply:
The only problem I see with this approach is when we get a combination of status, PartialTrace and smth else. We can end doing bitmasking :P